mne/utils.py

```diff
@@ -2193,13 +2193,35 @@ def _check_preload(inst, msg):
                 '%s.load_data().' % (name, name))
 
 
-def _check_compensation_grade(inst, inst2, name, name2):
-    """Ensure that objects have same compenstation_grade."""
-    if (None not in [inst.info, inst2.info]) and (inst.compensation_grade !=
-                                                  inst2.compensation_grade):
-            msg = ('Compensation grade of %s (%d) and %s (%d) don\'t match')
-            raise RuntimeError(msg % (name, inst.compensation_grade,
-                                      name2, inst2.compensation_grade))
+def _check_compensation_grade(inst, inst2, name, name2, ch_names=None):
+    """Ensure that objects have same compensation_grade."""
+    from .io.pick import pick_channels, pick_info
+    from .io.compensator import get_current_comp
+
+    if None in [inst.info, inst2.info]:
```

Comment:
When do we have `inst.info is None`?

Reply:
`ica` objects have a `info is None` until they are fit.

this is needed to this part of the tests pass:
https://github.com/mne-tools/mne-python/blob/68084ccfd8114d496bd64d28bb4f10bf792239c9/mne/preprocessing/tests/test_ica.py#L250-L252

An alternative would be to move the fit checking out the private `_transform_*` methods but that would likely be two checks as I think it makes sense to have it in the private method aswell since they are used in a number of places.